exec/java-exec/src/main/java/org/apache/drill/exec/client/DrillClient.java

```diff
@@ -74,81 +78,173 @@
+    this(newBuilder());
   }
 
-  public DrillClient(boolean isDirect) throws OutOfMemoryException {
-    this(DrillConfig.create(), isDirect);
+  /**
+   * @deprecated Create a DrillClient using {@link DrillClient.Builder}.
+   */
+  @Deprecated
+  public DrillClient(boolean isDirect) {
+    this(newBuilder()
+        .setDirectConnection(isDirect));
   }
 
-  public DrillClient(String fileName) throws OutOfMemoryException {
-    this(DrillConfig.create(fileName), false);
+  /**
+   * @deprecated Create a DrillClient using {@link DrillClient.Builder}.
+   */
+  @Deprecated
+  public DrillClient(String fileName) {
+    this(newBuilder()
+        .setConfigFromFile(fileName));
   }
 
-  public DrillClient(DrillConfig config) throws OutOfMemoryException {
-    this(config, null, false);
+  /**
+   * @deprecated Create a DrillClient using {@link DrillClient.Builder}.
+   */
+  @Deprecated
+  public DrillClient(DrillConfig config) {
+    this(newBuilder()
+        .setConfig(config));
   }
 
-  public DrillClient(DrillConfig config, boolean isDirect)
-      throws OutOfMemoryException {
-    this(config, null, isDirect);
+  /**
+   * @deprecated Create a DrillClient using {@link DrillClient.Builder}.
+   */
+  @Deprecated
+  public DrillClient(DrillConfig config, boolean isDirect) {
+    this(newBuilder()
+        .setConfig(config)
+        .setDirectConnection(isDirect));
   }
 
-  public DrillClient(DrillConfig config, ClusterCoordinator coordinator)
-    throws OutOfMemoryException {
-    this(config, coordinator, null, false);
+  /**
+   * @deprecated Create a DrillClient using {@link DrillClient.Builder}.
+   */
+  @Deprecated
+  public DrillClient(DrillConfig config, ClusterCoordinator coordinator) {
+    this(newBuilder()
+        .setConfig(config)
+        .setClusterCoordinator(coordinator));
   }
 
-  public DrillClient(DrillConfig config, ClusterCoordinator coordinator, boolean isDirect)
-    throws OutOfMemoryException {
-    this(config, coordinator, null, isDirect);
+  /**
+   * @deprecated Create a DrillClient using {@link DrillClient.Builder}.
+   */
+  @Deprecated
+  public DrillClient(DrillConfig config, ClusterCoordinator coordinator, boolean isDirect) {
+    this(newBuilder()
+        .setConfig(config)
+        .setClusterCoordinator(coordinator)
+        .setDirectConnection(isDirect));
   }
 
-  public DrillClient(DrillConfig config, ClusterCoordinator coordinator, BufferAllocator allocator)
-      throws OutOfMemoryException {
-    this(config, coordinator, allocator, false);
+  /**
+   * @deprecated Create a DrillClient using {@link DrillClient.Builder}.
+   */
+  @Deprecated
+  public DrillClient(DrillConfig config, ClusterCoordinator coordinator, BufferAllocator allocator) {
+    this(newBuilder()
+        .setConfig(config)
+        .setClusterCoordinator(coordinator)
+        .setAllocator(allocator));
+  }
+
+  /**
+   * @deprecated Create a DrillClient using {@link DrillClient.Builder}.
+   */
+  @Deprecated
```

Comment:
Hmm, I take that back, on reflection. Some apps might be using this class. Probably good idea to log a JIRA to remove these. in future.